### PR TITLE
feat: Added Opensearch 1.2.0

### PR DESCRIPTION
**Pull request by syedimranhassan**
----
https://openedx.atlassian.net/browse/ISRE-1306
Adding OpenSearch for LMS, studio, cs_comments, and blockstore.
Our plan is to replace ES with OS for these services.

I've completed each of the following or determined they are not applicable:

- [ ] Made a plan to communicate any major developer interface changes (or N/A)

### CODE REVIEW

**Review by NIXKnight — APPROVED** [review]
LGTM

File: docker-compose.yml [hunk 1]
```diff
@@ -418,6 +438,7 @@ services:
       - memcached
       - mongo
       - mysql57
+      - opensearch12
```
Comment (robrap) [inline docker-compose.yml:441]: 1. Can you update the PR description with a little more background about how and when each of the services will rely on this?
2. Will this just be needed for a rare use case? I’m not clear on what “test OS upgrades” entails, but it seems like a more rare activity and that maybe this isn’t a required dependency for most uses of LMS or other services? This is a question. If not, maybe there should simply be doc changes for when and how to bring this container up?

Comment (dianakhuang) [reply docker-compose.yml:441]: I think the best reference for this is the Elasticsearch -> OpenSearch ticket we have on the DEPR board. We are migrating all our services off Elasticsearch and for those that can't use MySQL full text search, we're going to use OpenSearch. https://github.com/openedx/public-engineering/issues/16

Comment (syedimranhassan) [reply docker-compose.yml:441]: Yeah, we are migrating LMS, studio, cs_comments(forum), and blockstore from Elasticsearch -> OpenSearch.

Comment (robrap) [reply docker-compose.yml:441]: Ok. Seems fine if you think we need both dependencies now.

**Review by robrap — APPROVED** [review]

**Pull request merged**